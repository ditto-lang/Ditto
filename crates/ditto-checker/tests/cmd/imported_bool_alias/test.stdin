module Test exports (..);

import Boolean (Boolean);

my_boolean : Boolean = true;

five = if Boolean.example then 5 else 5;