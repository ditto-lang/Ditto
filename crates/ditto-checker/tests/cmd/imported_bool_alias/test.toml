bin.name = "ditto-checker-testbin-check-module"
args = ["imported_bool_alias.ditto"]
fs.sandbox = true
