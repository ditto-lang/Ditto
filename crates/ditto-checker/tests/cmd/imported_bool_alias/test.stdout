{
  "module_name": [
    "Test"
  ],
  "exports": {
    "types": {},
    "constructors": {},
    "values": {
      "five": {
        "doc_comments": [],
        "doc_position": 0,
        "value_type": {
          "type": "PrimConstructor",
          "data": "Int"
        }
      },
      "my_boolean": {
        "doc_comments": [],
        "doc_position": 1,
        "value_type": {
          "type": "ConstructorAlias",
          "data": {
            "constructor_kind": "Type",
            "canonical_value": {
              "module_name": [
                null,
                [
                  "Boolean"
                ]
              ],
              "value": "Boolean"
            },
            "source_value": {
              "module_name": null,
              "value": "Boolean"
            },
            "alias_variables": [],
            "aliased_type": {
              "type": "PrimConstructor",
              "data": "Bool"
            }
          }
        }
      }
    }
  },
  "types": {},
  "constructors": {},
  "values": {
    "five": {
      "doc_comments": [],
      "name_span": {
        "start_offset": 81,
        "end_offset": 85
      },
      "expression": {
        "expression": "If",
        "data": {
          "span": {
            "start_offset": 88,
            "end_offset": 120
          },
          "output_type": {
            "type": "PrimConstructor",
            "data": "Int"
          },
          "condition": {
            "expression": "ImportedVariable",
            "data": {
              "span": {
                "start_offset": 91,
                "end_offset": 106
              },
              "variable_type": {
                "type": "ConstructorAlias",
                "data": {
                  "constructor_kind": "Type",
                  "canonical_value": {
                    "module_name": [
                      null,
                      [
                        "Boolean"
                      ]
                    ],
                    "value": "Boolean"
                  },
                  "source_value": {
                    "module_name": null,
                    "value": "Boolean"
                  },
                  "alias_variables": [],
                  "aliased_type": {
                    "type": "PrimConstructor",
                    "data": "Bool"
                  }
                }
              },
              "variable": {
                "module_name": [
                  null,
                  [
                    "Boolean"
                  ]
                ],
                "value": "example"
              }
            }
          },
          "true_clause": {
            "expression": "Int",
            "data": {
              "span": {
                "start_offset": 112,
                "end_offset": 113
              },
              "value": "5",
              "value_type": {
                "type": "PrimConstructor",
                "data": "Int"
              }
            }
          },
          "false_clause": {
            "expression": "Int",
            "data": {
              "span": {
                "start_offset": 119,
                "end_offset": 120
              },
              "value": "5",
              "value_type": {
                "type": "PrimConstructor",
                "data": "Int"
              }
            }
          }
        }
      }
    },
    "my_boolean": {
      "doc_comments": [],
      "name_span": {
        "start_offset": 52,
        "end_offset": 62
      },
      "expression": {
        "expression": "True",
        "data": {
          "span": {
            "start_offset": 75,
            "end_offset": 79
          },
          "value_type": {
            "type": "ConstructorAlias",
            "data": {
              "constructor_kind": "Type",
              "canonical_value": {
                "module_name": [
                  null,
                  [
                    "Boolean"
                  ]
                ],
                "value": "Boolean"
              },
              "source_value": {
                "module_name": null,
                "value": "Boolean"
              },
              "alias_variables": [],
              "aliased_type": {
                "type": "PrimConstructor",
                "data": "Bool"
              }
            }
          }
        }
      }
    }
  },
  "values_toposort": [
    "five",
    "my_boolean"
  ]
}